packages/create-cloudflare/src/templateMap.ts

```diff
@@ -1,53 +1,358 @@
+
+	path?: string;
+
+	testFlags?: string[];
+	compatibilityFlags?: string[];
+	deployCommand?: string[];
+	devCommand?: string[];
 };
 
-export const templateMap: Record<string, TemplateConfig> = {
-	"hello-world": {
-		label: `"Hello World" Worker`,
-		handler: runWorkersGenerator,
-	},
-	webFramework: {
-		label: "Website or web app",
-		handler: runPagesGenerator,
-	},
-	common: {
-		label: "Example router & proxy Worker",
-		handler: runWorkersGenerator,
-	},
-	scheduled: {
-		label: "Scheduled Worker (Cron Trigger)",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	queues: {
-		label: "Queue consumer & producer Worker",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	chatgptPlugin: {
-		label: `ChatGPT plugin`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	openapi: {
-		label: `OpenAPI 3.1`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	"pre-existing": {
-		label: "Pre-existing Worker (from Dashboard)",
-		handler: runWorkersGenerator,
-		hidden: true,
-	},
+// A template can have a number of variants, usually js/ts
+type VariantInfo = {
+	path: string;
+};
+
+type StaticFileMap = Record<string, VariantInfo>;
+
+export type FrameworkMap = Awaited<ReturnType<typeof getFrameworkMap>>;
+export type FrameworkName = keyof FrameworkMap;
+
+export const getFrameworkMap = async () => ({
+	angular: (await import("../templates/angular/c3")).default,
+	astro: (await import("../templates/astro/c3")).default,
+	docusaurus: (await import("../templates/docusaurus/c3")).default,
+	gatsby: (await import("../templates/gatsby/c3")).default,
+	hono: (await import("../templates/hono/c3")).default,
+	next: (await import("../templates/next/c3")).default,
+	nuxt: (await import("../templates/nuxt/c3")).default,
+	qwik: (await import("../templates/qwik/c3")).default,
+	react: (await import("../templates/react/c3")).default,
+	remix: (await import("../templates/remix/c3")).default,
+	solid: (await import("../templates/solid/c3")).default,
+	svelte: (await import("../templates/svelte/c3")).default,
+	vue: (await import("../templates/vue/c3")).default,
+});
+
+export const getTemplateMap = async () => {
+	return {
+		"hello-world": (await import("../templates/hello-world/c3")).default,
+		// Dummy record -- actual template config resolved in `selectFramework`
+		webFramework: { displayName: "Website or web app" } as TemplateConfig,
+		common: (await import("../templates/common/c3")).default,
+		scheduled: (await import("../templates/scheduled/c3")).default,
+		queues: (await import("../templates/queues/c3")).default,
+		chatgptPlugin: (await import("../templates/chatgptPlugin/c3")).default,
+		openapi: (await import("../templates/openapi/c3")).default,
+		// Dummy record -- actual template config resolved in `processRemoteTemplate`
+		"remote-template": {
+			displayName: "From a remote git repository",
```

Comment:
NIT: Is `From a remote git repository` correct?

![Screenshot 2024-01-19 at 01 11 41](https://github.com/cloudflare/workers-sdk/assets/61631103/69e5ceda-0566-461d-b4c1-e7be02985a96)

`From a remote git repository` is not a type of application 🤔 

```suggestion
			displayName: "Worker built from a remote git repository",
```